src/utils/sockctl.c

```diff
@@ -96,6 +96,33 @@ int create_domain_server(const char *server_path) {
   return sfd;
 }
 
+int close_domain_socket(int unix_domain_socket_fd) {
+  struct sockaddr_un sockaddr = {0};
+  socklen_t address_len = sizeof(sockaddr);
+  if (getsockname(unix_domain_socket_fd, &sockaddr, &address_len)) {
+    log_errno("Failed to getsockname for unix domain socket %d",
+              unix_domain_socket_fd);
+    return -1;
+  }
+  if (sockaddr.sun_family != AF_UNIX) {
+    log_error("Socket %d is not a unix domain socket, instead it's a %d",
+              unix_domain_socket_fd, sockaddr.sun_family);
+    return -1;
+  }
+  if (address_len >=
```

Comment:
Can you define a function for instance `is_unix_domain_socket` that returns 1 if it is and 0 if it is not and -1 on error?

Reply:
You are right. It is not worth it.

Reply:
Yeah, that's the annoying bit about C. Because you need to explicitly handle every return code, and almost every single function can return an error, little helper functions are sometimes not worth it (although I've heard Rust has similar error handling semantics).